src/dftbp/timedep/linrespcommon.F90

```diff
@@ -411,469 +406,464 @@ subroutine getSqrOcc(occNr, win, nmatup, nmat, getIA, tSpin, n_ij)
+    if (.not. lr%tSpin) then !-----------spin-unpolarized systems--------------
 
       if (sym == 'S') then
 
-        call hemv(gtmp, gamma, otmp)
+        call hemv(gtmp, frGamma, otmp)
 
         ! 4 * wn * (g * Q)
         vOut(:) = 0.0_dp
-        call transChrg%qVecMat(env, denseDesc, ovrXev, grndEigVecs, getIA, win, gTmp, vOut)
-        vOut(:) = 4.0_dp * sqrOccIA * vOut
+        do ia = iGlobal, fGlobal
+          myia = ia - iGlobal + 1
+          qij(:) = transChrg%qTransIA(ia, env, denseDesc, ovrXev, grndEigVecs, rpa%getIA, rpa%win)
+          vOut(myia) = 4.0_dp * rpa%sqrOccIA(ia) * dot_product(qij, gTmp)
+        enddo  
 
       else
-
-        otmp = otmp * spinW(species0)
+       
+        otmp = otmp * lr%spinW(species0)
 
         ! 4 * wn * (o * Q)
         vOut = 0.0_dp
-        call transChrg%qVecMat(env, denseDesc, ovrXev, grndEigVecs, getIA, win, oTmp, vOut)
-        vOut(:) = 4.0_dp * sqrOccIA * vOut
+        call transChrg%qVecMat(env, denseDesc, ovrXev, grndEigVecs, rpa%getIA, rpa%win, oTmp, vOut,&
+            & iGlobal-1)
+        vOut(:) = 4.0_dp * rpa%sqrOccIA(iGlobal:fGlobal) * vOut
 
 
       end if
 
     else !--------------spin-polarized systems--------
 
-      call hemv(gtmp, gamma, otmp)
+      call hemv(gtmp, frGamma, otmp)
 
       otmp(:) = 0.0_dp
 
-      !$OMP PARALLEL DO DEFAULT(SHARED) PRIVATE(ia,ss,qij) SCHEDULE(RUNTIME) REDUCTION(+:otmp)
-      do ia = 1, nmat
-
-        qij(:) = transChrg%qTransIA(ia, env, denseDesc, ovrXev, grndEigVecs, getIA, win)
+      do ia = iGlobal, fGlobal
+        myia = ia - iGlobal + 1
+        
+        qij(:) = transChrg%qTransIA(ia, env, denseDesc, ovrXev, grndEigVecs, rpa%getIA, rpa%win)
 
         ! singlet gamma part (S)
-        vout(ia) = 2.0_dp * sqrOccIA(ia) * dot_product(qij, gtmp)
+        vOut(myia) = 2.0_dp * rpa%sqrOccIA(ia) * dot_product(qij, gtmp)
 
         ! magnetization part (T1)
-        ss = getIA(win(ia), 3)
+        ss = rpa%getIA(rpa%win(ia), 3)
 
-        otmp(:) = otmp + spinFactor(ss) * sqrOccIA(ia) * vTmp(ia) * qij
+        otmp(:) = otmp + spinFactor(ss) * rpa%sqrOccIA(ia) * vLoc(myia) * qij
 
       end do
-      !$OMP END PARALLEL DO
 
-      otmp = otmp * spinW(species0)
+      call assembleChunks(env, otmp)
 
-      !$OMP PARALLEL DO DEFAULT(SHARED) PRIVATE(ia,ss,qij) SCHEDULE(RUNTIME)
-      do ia = 1,nmat
+      otmp = otmp * lr%spinW(species0)
 
-        qij(:) = transChrg%qTransIA(ia, env, denseDesc, ovrXev, grndEigVecs, getIA, win)
-
-        ss = getIA(win(ia), 3)
+      do ia = iGlobal, fGlobal
+        myia = ia - iGlobal + 1
+        
+        qij(:) = transChrg%qTransIA(ia, env, denseDesc, ovrXev, grndEigVecs, rpa%getIA, rpa%win)
 
-        vout(ia) = vout(ia) + 2.0_dp * sqrOccIA(ia) * spinFactor(ss) * dot_product(qij, otmp)
+        ss = rpa%getIA(rpa%win(ia), 3)
+      
+        vOut(myia) = vOut(myia) + 2.0_dp * rpa%sqrOccIA(ia) * spinFactor(ss) * &
+            & dot_product(qij, otmp)
```

Comment:
```suggestion
        vOut(myia) = vOut(myia) &
            & + 2.0_dp * rpa%sqrOccIA(ia) * spinFactor(ss)  * dot_product(qij, otmp)
```
It is easier to read, if you put the arithmetic operator at the beginning of the continuation line. And if easily possible, break the expression according to the mathematical priority of the operators.